File: src/server/TypeLibrary.Api/Terminals/TerminalView.cs
using TypeLibrary.Api.Common;
using TypeLibrary.Core.Common;
using TypeLibrary.Core.Terminals;

namespace TypeLibrary.Api.Terminals;

public class TerminalView : ImfType
{
    public ICollection<RdlClassifier> Classifiers { get; set; } = new List<RdlClassifier>();
    public RdlPurpose? Purpose { get; set; }
    public string? Notation { get; set; }
    public string? Symbol { get; set; }
    public Aspect? Aspect { get; set; }
    public RdlMedium? Medium { get; set; }
    public Direction Qualifier { get; set; }
    public ICollection<AttributeTypeReferenceView> Attributes { get; set; } = new List<AttributeTypeReferenceView>();
}

File: src/client/src/common/types/terminals/terminalView.ts
import { Aspect } from "../common/aspect";
import { AttributeTypeReferenceView } from "../common/attributeTypeReferenceView";
import { ImfType } from "../common/imfType";
import { RdlClassifier } from "../common/rdlClassifier";
import { RdlPurpose } from "../common/rdlPurpose";
import { Direction } from "./direction";
import { RdlMedium } from "./rdlMedium";

export interface TerminalView extends ImfType {
  classifiers: RdlClassifier[];
  purpose?: RdlPurpose;
  notation?: string;
  symbol?: string;
  aspect?: Aspect;
  medium?: RdlMedium;
  direction: Direction;
  attributes: AttributeTypeReferenceView[];
  minCount: number;
  maxCount?: number;
}
